### ISSUE TITLE
Unused or unrecognized kwargs: padding, max_length, truncation.

### ISSUE THREAD

**Issue by Excuses123** [huggingface/transformers#30106]
### System Info

- `transformers` version: 4.39.2
- Platform: Linux-3.10.0-1160.99.1.el7.x86_64-x86_64-with-glibc2.31
- Python version: 3.10.13
- Huggingface_hub version: 0.22.2
- Safetensors version: 0.4.2
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Fine-tuning the CLIP model. Previously, I passed both text and images to the processor and specified some parameters for the text: padding, max_length, truncation. Everything was working fine before upgrading to version 4.39. However, after the upgrade, I encountered a warning that affects the training process. How should I resolve this? 
![image](https://github.com/huggingface/transformers/assets/22993056/2d8d8ece-f190-4432-bcfe-43ea3439c848)

### Expected behavior

Do I need to separate the text and images for processing and use processor.tokenizer and processor.image_processor separately?

**Comment by kuri54** [comment]
Same problem is occurring.
Downgrading to 4.38.2 will solve the problem.

**Comment by amyeroberts** [comment]
Hi @Excuses123, thanks for raising this issue! 

Could you share a minimal reproducer?

**Comment by hda-xian** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by Excuses123** [comment]
> Hi @Excuses123, thanks for raising this issue!
> 
> Could you share a minimal reproducer?

Hello, here is a minimal reproducible example.

import torch
import requests
from PIL import Image
from transformers import AutoProcessor

processor = AutoProcessor.from_pretrained("openai/clip-vit-large-patch14-336")

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

processor(text="a photo of a cat", 
          images=image, 
          max_length=64, 
          padding='max_length', 
          truncation=True, 
          return_tensors='pt')

**Comment by amyeroberts** [comment]
Thanks for sharing @Excuses123. The warnings have started because of new input validation on the image processor. This is good as it shows tokenizer kwargs were being passed to the image processor when they shouldn't be! I've opened a PR to address this

**Comment by hda-xian** [comment]
just to Change code
for the /**your paht****/transformers/models/clip/processing_clip.py 
     function : def __call__(self, text=None, images=None, return_tensors=None, **kwargs):
     line:  104   image_features = self.image_processor(images, return_tensors=return_tensors, **kwargs)
    the new code :  ->  image_features = self.image_processor(images, return_tensors=return_tensors)  # delete the others kwargs
